pkg/core/mempool/mem_pool.go

```diff
@@ -324,13 +314,22 @@ func (mp *Pool) removeInternal(hash util.Uint256) {
 	} else if num == len(mp.verifiedTxes)-1 {
 		mp.verifiedTxes = mp.verifiedTxes[:num]
 	}
+	mp.removeFromMapWithFeesAndAttrs(itm)
+}
+
+// removeFromMapWithFeesAndAttrs removes given item (with the given hash) from
+// verifiedMap, adjusts fees, handles attributes and fires and event. Notice
+// that it does not do anything to verifiedTxes (the presumption is that if
+// you have itm already, you can handle it fine for the specific case).
+func (mp *Pool) removeFromMapWithFeesAndAttrs(itm item) {
```

Comment:
May we add a couple of words about the fact that this method is unlocked and it's a caller's duty to take an appropriate mutex?